admin/README.md

```diff
@@ -0,0 +1,58 @@
+## Tag Driven Releasing
+
+### Background Reading
+
+  - http://docs.travis-ci.com/user/environment-variables/
+  - http://docs.travis-ci.com/user/encryption-keys/
+  - http://docs.travis-ci.com/user/encrypting-files/
+
+### Initial setup for the repository
+
+To configure tag driven releases from Travis CI.
+
+  1. Generate a key pair for this repository with `./admin/genKeyPair.sh`.
+     Edit `.travis.yml` and `admin/build.sh` as prompted.
+  2. Publish the public key to https://pgp.mit.edu
+  3. Store other secrets as encrypted environment variables with `admin/encryptEnvVars.sh`.
+     Edit `.travis.yml` as prompted.
+  4. Edit `.travis.yml` to use `./admin/build.sh` as the build script,
+     and edit that script to use the tasks required for this project.
+  5. Edit `.travis.yml` to select which JDK will be used for publishing.
+
+It is important to add comments in .travis.yml to identify the name
+of each environment variable encoded in a `:secure` section.
+
+After all of these steps, your .travis.yml should contain config of the
+form:
+
+	language: scala
+	env:
+	  global:
+	    - PUBLISH_JDK=openjdk6
+	    # PGP_PASSPHRASE
+	    - secure: "XXXXXX"
+	    # SONA_USER
+	    - secure: "XXXXXX"
+	    # SONA_PASS
+	    - secure: "XXXXXX"
```

Comment:
Here's the log of running `genKeyPair.sh`

```
sh -x ./admin/genKeyPair.sh
+ set -e
+ for f in admin/secring.asc.enc admin/secring.asc admin/pubring.asc
+ promptDelete admin/secring.asc.enc
+ [[ -f admin/secring.asc.enc ]]
+ echo About to delete admin/secring.asc.enc, Enter for okay / CTRL-C to cancel
About to delete admin/secring.asc.enc, Enter for okay / CTRL-C to cancel
+ read

+ rm admin/secring.asc.enc
+ for f in admin/secring.asc.enc admin/secring.asc admin/pubring.asc
+ promptDelete admin/secring.asc
+ [[ -f admin/secring.asc ]]
+ for f in admin/secring.asc.enc admin/secring.asc admin/pubring.asc
+ promptDelete admin/pubring.asc
+ [[ -f admin/pubring.asc ]]
+ echo About to delete admin/pubring.asc, Enter for okay / CTRL-C to cancel
About to delete admin/pubring.asc, Enter for okay / CTRL-C to cancel
+ read

+ rm admin/pubring.asc
+ echo Generating key pair. Please enter 1. repo name 2. scala-internals@googlegroups.com, 3. a new passphrase
Generating key pair. Please enter 1. repo name 2. scala-internals@googlegroups.com, 3. a new passphrase
+ cp admin/gpg.sbt project
+ sbt 'set pgpReadOnly := false' 'set pgpPublicRing := file("admin/pubring.asc")' 'set pgpSecretRing := file("admin/secring.asc")' 'pgp-cmd gen-key'
[info] Loading global plugins from /Users/jason/.sbt/0.13/plugins
[info] Loading project definition from /Users/jason/code/scala-java8-compat/project/project
[info] Loading project definition from /Users/jason/code/scala-java8-compat/project
[warn] There may be incompatibilities among your library dependencies.
[warn] Here are some of the libraries that were evicted:
[warn] 	* com.typesafe.sbt:sbt-pgp:0.8.1 -> 0.8.3
[warn] Run 'evicted' to see detailed eviction warnings
[info] Set current project to scala-java8-compat (in build file:/Users/jason/code/scala-java8-compat/)
[info] Defining */*:pgpReadOnly
[info] The new value will be used by no settings or tasks.
[info] Reapplying settings...
[info] Set current project to scala-java8-compat (in build file:/Users/jason/code/scala-java8-compat/)
[info] Defining */*:pgpPublicRing
[info] The new value will be used by */*:pgpStaticContext
[info] Reapplying settings...
[info] Set current project to scala-java8-compat (in build file:/Users/jason/code/scala-java8-compat/)
[info] Defining */*:pgpSecretRing
[info] The new value will be used by */*:pgpStaticContext
[info] Reapplying settings...
[info] Set current project to scala-java8-compat (in build file:/Users/jason/code/scala-java8-compat/)
Please enter the name associated with the key: scala-java8-compat
Please enter the email associated with the key: scala-internals@googlegroups.com
Please enter the passphrase for the key: ***************
Please re-enter the passphrase for the key: ***************
[info] Creating a new PGP key, this could take a long time.
[info] Public key := /Users/jason/code/scala-java8-compat/admin/pubring.asc
[info] Secret key := /Users/jason/code/scala-java8-compat/admin/secring.asc
[info] Please do not share your secret key.   Your public key is free to share.
+ rm project/gpg.sbt
+ echo ============================================================================================
============================================================================================
+ echo Encrypting admin/secring.asc. Update K and IV variables in admin/build.sh accordingly.
Encrypting admin/secring.asc. Update K and IV variables in admin/build.sh accordingly.
+ echo ============================================================================================
============================================================================================
+ travis encrypt-file admin/secring.asc
encrypting admin/secring.asc for scala/scala-java8-compat
storing result as secring.asc.enc
storing secure env variables for decryption

Please add the following to your build script (before_install stage in your .travis.yml, for instance):

    openssl aes-256-cbc -K $encrypted_1ce132863fa7_key -iv $encrypted_1ce132863fa7_iv -in secring.asc.enc -out admin/secring.asc -d

Pro Tip: You can add it automatically by running with --add.

Make sure to add secring.asc.enc to the git repository.
Make sure not to add admin/secring.asc to the git repository.
Commit all changes to your .travis.yml.
+ rm admin/secring.asc
+ mv secring.asc.enc admin
+ echo ============================================================================================
============================================================================================
+ echo Encrypting environment variables. Add each to a line in .travis.yml. Include a comment
Encrypting environment variables. Add each to a line in .travis.yml. Include a comment
+ echo with the name of the corresponding variable
with the name of the corresponding variable
+ echo ============================================================================================
============================================================================================
+ read -s -p 'PGP_PASSPHRASE: ' PGP_PASSPHRASE
PGP_PASSPHRASE: + travis encrypt PGP_PASSPHRASE=D1cXBjsw15dLUA4
Please add the following to your .travis.yml file:

  secure: "YwLY+gZ1TQvRYEGMNKIHR6RAinI4UftTpXGEESQqOGIZvfKW3/rZqMGVU1vnEtgBQENH7jrL+ujFsH0K4t36WF3jgnGd/fYhiODuOofi0+ZWG0hcvt0EtM9XguvT7BrmWn1+zTdVuZyz8IYxacVpk6sToDyLwxmjUdGMEMu5qV8="
```